include/arch/riscv/arch/model/smp.h

```diff
@@ -51,7 +51,8 @@ static inline bool_t try_arch_atomic_exchange_rlx(void *ptr, void *new_val, void
 
 static inline CONST cpu_id_t getCurrentCPUIndex(void)
```

Comment:
What about just pulling in the prototype here:

```suggestion
/* We can't include arch/machine.h due because there is a circular include
 * dependency. Until that can be resolved, we need to define the function
 * prototype here.
 */
 
static inline word_t read_sscratch(void);
static inline CONST cpu_id_t getCurrentCPUIndex(void)
```

Reply:
We could do this also, but I don't see a real gain here compared to just using the prototype. The compiler will warn if the prototype does no longer match and we have the comment that explains things. Te long term solution would be refactoring things then to resolve the include quirk, but seem this is not trivial.